test/Main.purs

```diff
@@ -31,6 +31,13 @@ main = do
     log "`genNonEmpty` should not reduce the remainder size below zero"
     one :: NonEmpty Array Int ← Gen.resize (const 0) $ GenC.genNonEmpty (Gen.sized pure)
     liftEffect $ assertEqual { actual: one, expected: 0 :| [] }
+    
+    log "Ensure that `elements` will produce all possible values (tests will hang if this fails)"
+    _ ← Gen.suchThat (Gen.elements ("A" :| ["B", "C", "D"])) (_ == "A")
```

Comment:
Couldn't we do something that will fail (rather than hang CI for who knows how long) like this?
```purescript
test = do
  let testElems = "A" :| ["B", "C", "D"]
  elems <- randomSample' 100 (Gen.elements testElems)
  let generatedElems = foldl (\mymap e -> Map.insertWith (_ + 1) e 0 mymap) Map.empty elems
  keys generatedElems `shouldEqual` (toArray testElems)
  -- some other frequency test here, too.

  
```

Reply:
Only if we don't mind there occasionally being random failures. There's no limited number of samples you can generate that won't result in a false negative from time to time.

I suppose that's still technically true with the current test, just the number of things it will generate is bound by the CI timeout or node crashing rather than there being an explicit limit.

Reply:
> Only if we don't mind there occasionally being random failures. There's no limited number of samples you can generate that won't result in a false negative from time to time.

True, but at least CI fails quicker. Also, if we increase the number of elements generated, wouldn't that decrease the possibility of a random failure occurring?

Reply:
Yeah, the larger the number the less likely it will fail.

Reply:
So.... 1,000, right? :laughing:

Reply:
I'll merge this PR and then submit a PR that uses the above approach.